### DISCUSSION TITLE
Does aspire only use for develop now ?

### DISCUSSION THREAD

**Discussion by ohroy** [microsoft/aspire#1594] [General]
It provider Dashboard, Service Discovery, Health Check, Telemetry, All of this is usually very useful in a production environment, So I think    aspire can use for produdction, but How can it boot the project when production? 

The dotnet app can delopy with many ways, such as container, every service has it's own container, then How can aspire boot it?

If it can't use for production, Why services need `AddServiceDefaults` ?

**Comment by davidfowl** [comment]
"production" is an overloaded term so I'll use the term "deployable" (as you deploy to different environments). [azd can deploy an aspire application to azure container apps](https://learn.microsoft.com/en-us/dotnet/aspire/deployment/azure/aca-deployment-azd-in-depth?tabs=windows), and [aspir8 can deploy an aspire application to Kubernetes](https://github.com/prom3theu5/aspirational-manifests) (there'll be support for AWS tooling as well). These will be our primary deployment targets for aspire v1. Work is being done to make the runnable outside of the local environment so that it can be deployed (https://github.com/dotnet/aspire/issues/1003).

**Comment by ohroy** [comment]
Sorry my bad, my english is poor.
I understand this tool by read its source code now, in this moment, I think it's just a very useful Orchestration tool. all of the extra functional like Service Discovery, Health Check, Telemetry are pair with dashboard, but the dashboard is standing with develop...

So, now, the clean way to use this tool, is DO NOT use `AddServiceDefaults` in production env (cause it's useless) before #1003 finished.

Am I right to think this way? Thanks!!!

**Comment by davidfowl** [comment]
> I understand this tool by read its source code now, in this moment, I think it's just a very useful Orchestration tool. 

The orchestrator is for development only. That's why tools like azd and aspire8 "target" orchestrators of their own. The apphost project is never deployed.

> all of the extra functional like Service Discovery, Health Check, Telemetry are pair with dashboard, but the dashboard is standing with develop...

The code in AddServiceDefaults isn't tied to the dashboard. [This example](https://github.com/Depechie/aspiresample) shows running aspire with the grafana telemetry stack, (loki and tempo for logs and distributed tracing). [This example](https://github.com/dotnet/aspire-samples/tree/main/samples/Metrics) shows how to wire up Prometheus. 

The service defaults are part of your app code and is a pattern for wiring up cross cutting concerns (like your services). Service discovery, resiliency and health checks have no reliance on the dashboard.

If you want to delete the service defaults then feel free to do so 😄

**Reply by ohroy** [reply]
Now the last piece of the puzzle is complete, you are my hero ! thanks !
